src/massbal.c

```diff
@@ -1085,72 +1086,104 @@ double massbal_getStoredMass(int p)
 
 //=============================================================================
 
-double massbal_getRoutingFlowTotal(int element)
+int massbal_getRoutingFlowTotal(int paramtype, double *value)
 //
 // Input:    element = element to return
 // Return:   value
 // Purpose:  Gets the routing total for toolkitAPI
 //
 {
-	double value;
+	int errorcode = 0;
 
-	switch (element)
+	// Check if Open
+	if (swmm_IsOpenFlag() == FALSE)
 	{
+		errorcode = ERR_API_INPUTNOTOPEN;
+	}
+
+	// Check if Simulation is Running
+	if (swmm_IsStartedFlag() == FALSE)
+	{
+		errorcode = ERR_API_SIM_NRUNNING;
+	}
+
+	if (errorcode == 0)
+	{
+		switch (paramtype)
+		{
 		// Cumulative Dry Weather Inflow Volume
-		case 0: value = FlowTotals.dwInflow * UCF(VOLUME); break;
+		case 0: *value = FlowTotals.dwInflow * UCF(VOLUME); break;
 		// Cumulative Wet Weather Inflow Volume
-		case 1: value = FlowTotals.wwInflow * UCF(VOLUME); break;
+		case 1: *value = FlowTotals.wwInflow * UCF(VOLUME); break;
 		// Cumulative Groundwater Inflow Volume
-		case 2: value = FlowTotals.gwInflow * UCF(VOLUME); break;
+		case 2: *value = FlowTotals.gwInflow * UCF(VOLUME); break;
 		// Cumulative I&I Inflow Volume
-		case 3: value = FlowTotals.iiInflow * UCF(VOLUME); break;
+		case 3: *value = FlowTotals.iiInflow * UCF(VOLUME); break;
 		// Cumulative External Inflow Volume
-		case 4: value = FlowTotals.exInflow * UCF(VOLUME); break;
+		case 4: *value = FlowTotals.exInflow * UCF(VOLUME); break;
 		// Cumulative Flooding Volume
-		case 5: value = FlowTotals.flooding * UCF(VOLUME); break;
+		case 5: *value = FlowTotals.flooding * UCF(VOLUME); break;
 		// Cumulative Outflow Volume
-		case 6: value = FlowTotals.outflow  * UCF(VOLUME); break;
+		case 6: *value = FlowTotals.outflow  * UCF(VOLUME); break;
 		// Cumulative Evaporation Loss
-		case 7: value = FlowTotals.evapLoss * UCF(VOLUME); break;
+		case 7: *value = FlowTotals.evapLoss * UCF(VOLUME); break;
 		// Cumulative Seepage Loss
-		case 8: value = FlowTotals.seepLoss * UCF(VOLUME); break;
+		case 8: *value = FlowTotals.seepLoss * UCF(VOLUME); break;
 		// Routing Error
-		case 9: value = massbal_getFlowError(); break;
+		case 9: *value = massbal_getFlowError(); break;
+		// default
+		default: errorcode = ERR_API_OUTBOUNDS; break;
```

Comment:
I would prefer a more descriptive error code. Something "Invalid parameter code" The more descriptive the error message the easier it is for the user receiving it to diagnose and solve the problem on their own. That's why it's important.